configure.ac

```diff
@@ -55,7 +55,7 @@ AC_CONFIG_HEADERS([config_ac.h])
 AM_CFLAGS='-Wall -I../config'
 AM_CPPFLAGS='-D_DEBUG -D_GNU_SOURCE -D_DEFAULT_SOURCE -DDMALLOC_DISABLE -DBUILD_LABEL=\""$(BUILD_LABEL)"\"'
 
-AC_DEFUN([AX_AM_CFLAGS_ADD],[AX_CHECK_COMPILE_FLAG($1, AM_CFLAGS="$AM_CFLAGS $1")])
+AC_DEFUN([AX_AM_CFLAGS_ADD],[AX_CHECK_COMPILE_FLAG($1, AM_CFLAGS="$AM_CFLAGS $1", , -Werror)])
 AX_AM_CFLAGS_ADD([-Wformat -Wformat-security -Werror])
```

Comment:
Do we need `-Werror` here?
I read manual: https://www.gnu.org/software/autoconf-archive/ax_check_compile_flag.html
I'm not sure what `EXTRA-FLAGS` brings here but I will not block to make it this way.

Reply:
> Do we need `-Werror` here? I read manual: https://www.gnu.org/software/autoconf-archive/ax_check_compile_flag.html I'm not sure what `EXTRA-FLAGS` brings here but I will not block to make it this way.

From the documentation "Check whether the given FLAG works with the current language’s compiler or gives an error. (**Warnings, however, are ignored**)".

When checking for flags that don't exist we get a warning, but the compiler switch is still included in the compile.  This will cause the build to fail as we are treating warnings as errors in the build which has nothing to do with this change.  Adding the `-Werror` does what we want in the check in that a compiler flag that doesn't work for a compiler isn't used during the compile.

Reply:
> I use clang periodically to check the code against different compiler implementations. Needed changes to get clang working again.

Should we consider adding clang to gh actions?

> When checking for flags that don't exist we get a warning,

Wow that is weird, I would expect error. Anyway, agree that we need to address it.

ohh, I see now. You did the trick to make warning as errors during flag verification, with the change not supported flags are not added.

I would say that change indirectly forces to use compiler with `-Werror` support but... maybe not if only "warning" is returned if it is not supported. I think that we should not care about compiler without `-Werror` support and we should just add it with no verification:
```
AM_CFLAGS='-Wall -I../config -Werror'
```
Your change with `-Werror` in EXTRA-FLAGS is good. It is about `-Werror` in line 59: `AX_AM_CFLAGS_ADD([-Wformat -Wformat-security -Werror]) ` , I don't see it valuable now.
Comments?

Reply:
>Your change with -Werror in EXTRA-FLAGS is good. It is about -Werror in line 59: AX_AM_CFLAGS_ADD([-Wformat -Wformat-security -Werror]) , I don't see it valuable now.
Comments?

I actually run `-Werror` on some other projects, but I can see reasons for not using it too.  We could simply enable it for CI testing, but otherwise have it not enabled.

Reply:
> Should we consider adding clang to gh actions?

IMHO it would be probably good to add it.  I can add it to this PR if you are good with it.

Reply:
> > Your change with -Werror in EXTRA-FLAGS is good. It is about -Werror in line 59: AX_AM_CFLAGS_ADD([-Wformat -Wformat-security -Werror]) , I don't see it valuable now.
> > Comments?
> 
> I actually run `-Werror` on some other projects, but I can see reasons for not using it too. We could simply enable it for CI testing, but otherwise have it not enabled.

Currently it is always enabled by:
```
AX_AM_CFLAGS_ADD([-Wformat -Wformat-security -Werror])
```
 I tested following code:
```diff
diff --git a/configure.ac b/configure.ac
index 915e5aa..a73db20 100644
--- a/configure.ac
+++ b/configure.ac
@@ -56,7 +56,7 @@ AM_CFLAGS='-Wall -I../config'
 AM_CPPFLAGS='-D_DEBUG -D_GNU_SOURCE -D_DEFAULT_SOURCE -DDMALLOC_DISABLE -DBUILD_LABEL=\""$(BUILD_LABEL)"\"'

 AC_DEFUN([AX_AM_CFLAGS_ADD],[AX_CHECK_COMPILE_FLAG($1, AM_CFLAGS="$AM_CFLAGS $1")])
-AX_AM_CFLAGS_ADD([-Wformat -Wformat-security -Werror])
+AX_AM_CFLAGS_ADD([-Wformat -Wformat-security])
 AX_AM_CFLAGS_ADD([-Wformat-overflow=2])
 AX_AM_CFLAGS_ADD([-Wno-strict-overflow])
 AX_AM_CFLAGS_ADD([-fno-delete-null-pointer-checks])
```

``` Bash
ledmon 0.97.0 configuration:

  Source code location:    .
  Preprocessor flags:      -D_DEBUG -D_GNU_SOURCE -D_DEFAULT_SOURCE -DDMALLOC_DISABLE -DBUILD_LABEL=\""$(BUILD_LABEL)"\"
  C compiler flags:        -Wall -I../config -Wformat -Wformat-security -Wformat-overflow=2 -Wno-strict-overflow -fno-delete-null-pointer-checks -Wformat-truncation=1 -Wshift-negative-value -Walloca -Wmissi
ng-field-initializers -Wformat-signedness -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fPIE
  Linker flags:             -pie -Wl,-z,relro -Wl,-z,now -Wl,-z,noexecstack
  Common install location: /usr
  configure parameters:    --enable-systemd=no --enable-library=no --enable-test=no --enable-doc=yes
                                                                                                                                                                                                              [gklab-localhost ~/mtkaczyk_work/ledmon/ledmon]# CFLAGS=-Werror
[gklab-localhost ~/mtkaczyk_work/ledmon/ledmon]# export CFLAGS=-Werror
[gklab-localhost ~/mtkaczyk_work/ledmon/ledmon]# ./configure
[...]
ledmon 0.97.0 configuration:

  Source code location:    .
  Preprocessor flags:      -D_DEBUG -D_GNU_SOURCE -D_DEFAULT_SOURCE -DDMALLOC_DISABLE -DBUILD_LABEL=\""$(BUILD_LABEL)"\"
  C compiler flags:        -Wall -I../config -Wformat -Wformat-security -Wformat-overflow=2 -Wno-strict-overflow -fno-delete-null-pointer-checks -Wformat-truncation=1 -Wshift-negative-value -Walloca -Wmissi
ng-field-initializers -Wformat-signedness -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fPIE -Werror
  Linker flags:             -pie -Wl,-z,relro -Wl,-z,now -Wl,-z,noexecstack
  Common install location: /usr
  configure parameters:    --enable-systemd=no --enable-library=no --enable-test=no --enable-doc=yes

```

I will open new pull to remove Werror from default flags and I will add it to CI next week.
Thanks,
Mariusz

Reply:
Update?  FYI: I didn't  get a notification when you did a thumbs up on my previous comment.  Otherwise I would have added a commit to this PR to enable clang testing in CI.

Reply:
Oh, sorry. Yes please add it.
I noted that I need to make pull, unfortunately scope on other task increased so I need to postpone working on this.
Hopefully, I will finish next week..

Reply:
@mtkaczyk Done